arguments-optional.js

```diff
@@ -0,0 +1,22 @@
+/**
+ *sums two arguments together. If only one argument is provided, returns a function that expects one argument and returns the sum. If either argument isn't a valid number, returns undefined.
+ *
+ * @param {*} a
+ * @param {*} b
+ * @returns {number|undefined} sum of a and b or undefined
+ */
+function addTogether(a, b) {
+  if (arguments.length === 2 && typeof(a, b) === "number") {
+    return a + b
+  } 
+  else if (arguments.length === 1 && typeof(a) === "number") {
+    return function addOther(c) {
```

Comment:
not sure why you need a `c` variable here.

Reply:
"If only one argument is provided, then return a function that expects one argument and returns the sum...Calling this returned function with a single argument will then return the sum" 

I did include in the description in the JSDoc part. Is that the right place for it?

Reply:
You don't need `c` variable. Look at code posted below.